cpinstaller/src/cpinstaller.js

```diff
@@ -55,32 +61,36 @@ export class CPInstallButton extends InstallButton {
         this.init();
     }
 
+    static get observedAttributes() {
+        return Object.keys(attrMap);
+    }
+
     connectedCallback() {
         this.boardName = this.getAttribute("boardname") || "ESP32-based device";
+        this.menuTitle = `CircuitPython Installer for ${this.boardName}`;
 
         // If this is empty, it's a problem
         this.boardId = this.getAttribute("boardid");
         this.releaseVersion = this.getAttribute("version");
 
-        // We need either the bootloader and uf2 or bin file to continue
-        this.bootloaderUrl = this.getBinaryUrl("bootloader");
-        this.uf2FileUrl = this.getBinaryUrl("uf2file");
-        this.binFileUrl = this.getBinaryUrl("binfile");
-
         // Nice to have for now
         this.chipFamily = this.getAttribute("chipfamily");
         this.bootloaderId = this.getAttribute("bootloaderid"); // This could be used to check serial output from board matches the UF2 file
 
         super.connectedCallback();
     }
 
-    getBinaryUrl(attribute) {
-        let url = this.getAttribute(attribute);
-        if (location.hostname == "localhost") {
+    attributeChangedCallback(attribute, previousValue, currentValue) {
+        const classVar = attrMap[attribute];
+        this[classVar] = currentValue ? this.updateBinaryUrl(currentValue) : null;
+    }
+
+    updateBinaryUrl(url) {
+        //if (location.hostname == "localhost") {
             if (url) {
                 url = url.replace("https://downloads.circuitpython.org/", "https://adafruit-circuit-python.s3.amazonaws.com/");
             }
-        }
+        //}
```

Comment:
Just checking that this is not leftover debugging commented-out stuff

Reply:
Correct. That's just so it will use Amazon S3 for the download on everything. Once we figure out why cloudfront keeps refusing to give the correct allow headers, we can revert it back by uncommenting.